pkg/cache/cache.go

```diff
@@ -721,6 +725,18 @@ func (c *Cache) MatchingClusterQueues(nsLabels map[string]string) sets.Set[strin
 	return cqs
 }
 
+// ShouldCheckWorkloadsPreemption returns true if it has workloads pending preemption now.
+func (c *Cache) ShouldCheckWorkloadsPreemption() bool {
+	c.RLock()
+	defer c.RUnlock()
+	for _, cq := range c.clusterQueues {
+		if now, _ := cq.GetPreemptingWorkloads(c.admissionChecks); len(now) > 0 {
```

Comment:
Can we do it only in the queue has any admissionchecks configured? If yes, that could reduce the number of workloads to be checked to only those using 2stage admission.

Reply:
Unfortunately no, well need to keep 2 ways of managing preemption in-place, also the list of checks of a cluster-queue could be changed at any point.